lib/p5.sound.js

```diff
@@ -6771,11 +6769,20 @@ effect = function () {
     this.output.disconnect();
     this.output = undefined;
   };
+  p5.Effect.prototype.dispose = function () {
+    var index = p5sound.soundArray.indexOf(this);
+    p5sound.soundArray.splice(index, 1);
+    this.input.disconnect();
+    this.input = undefined;
+    this.output.disconnect();
+    this.output = undefined;
```

Comment:
* Just did a quick reading on the 2 articles you've linked to.  
* They're very old. And they still had fears about the very old IE non-compliant behaviors.
* Nowadays most of those glitches are gone, especially when activating `"use strict";`.
* The rule is simple: if there's only 1 property referencing an object, if we reassign that property w/ something else, or outright `delete` the property itself, the now unreferenced object is gonna be collected by the GC eventually.